src/App.tsx

```diff
@@ -33,25 +40,62 @@ export const PATHS = {
   WELCOME_PAGE: "/welcome",
 };
 
+const queryClient = new QueryClient({
+  defaultOptions: {
+    mutations: {
+      networkMode: 'online',
+    },
+    queries: {
+      networkMode: 'online',
+      staleTime: 1000 * 20, // 20 seconds
+      gcTime: 1000 * 60 * 60 * 24 * 7, // 1 week
+      retry: 0,
+    }
+  }
+});
+addDefaultMutationFns(queryClient);
+
+const store = new Storage();
+await store.create();
+
+const persister = createAsyncStoragePersister({
+  storage: {
+    getItem: async (key) => store.get(key),
+    setItem: async (key, value) => store.set(key, value),
+    removeItem: async (key) => store.remove(key),
+  },
+});
+
 const App: React.FC = () => (
-  <IonApp>
-    <IonReactRouter>
-      <IonRouterOutlet>
-        <Route exact path={PATHS.WELCOME_PAGE}>
-          <WelcomePage />
-        </Route>
-        <Route exact path={PATHS.TUTORIAL_PAGE}>
-          <TutorialPage />
-        </Route>
-        <Route exact path={PATHS.HOME_PAGE}>
-          <Home />
-        </Route>
-        <Route exact path="/">
-          <Redirect to={PATHS.WELCOME_PAGE} />
-        </Route>
-      </IonRouterOutlet>
-    </IonReactRouter>
-  </IonApp>
+  <PersistQueryClientProvider
```

Comment:
The documentation for the `PersistQueryClientProvider` component is at: https://tanstack.com/query/latest/docs/framework/react/plugins/persistQueryClient#persistqueryclientprovider